pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -3,54 +3,114 @@ package opentsdb
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 	"path"
 	"strconv"
 	"strings"
 	"time"
 
-	"golang.org/x/net/context/ctxhttp"
-
 	"encoding/json"
-	"io/ioutil"
-	"net/http"
-	"net/url"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/infra/httpclient"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
+	"github.com/grafana/grafana/pkg/registry"
 	"github.com/grafana/grafana/pkg/setting"
+	"golang.org/x/net/context/ctxhttp"
 )
 
-type OpenTsdbExecutor struct {
-	httpClientProvider httpclient.Provider
+type Service struct {
+	HTTPClientProvider   httpclient.Provider   `inject:""`
+	Cfg                  *setting.Cfg          `inject:""`
+	BackendPluginManager backendplugin.Manager `inject:""`
+
+	im instancemgmt.InstanceManager
+}
+
+type datasourceInfo struct {
+	HTTPClient        *http.Client
+	Url               string `json:"url"`
+	BasicAuth         bool   `json:"basicAuth"`
+	BasicAuthUser     string `json:"basicAuthUser"`
+	BasicAuthPassword string `json:"basicAuthPassword"`
+}
+
+type DsAccess string
+
+func init() {
+	registry.Register(&registry.Descriptor{Instance: &Service{}})
+}
+
+func (s *Service) Init() error {
+	s.im = datasource.NewInstanceManager(NewInstanceSettings(s.HTTPClientProvider))
+
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: s,
+	})
+
+	if err := s.BackendPluginManager.Register("opentsdb", factory); err != nil {
+		plog.Error("Failed to register plugin", "error", err)
+	}
+
+	return nil
 }
 
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func New(httpClientProvider httpclient.Provider) func(*models.DataSource) (plugins.DataPlugin, error) {
-	//nolint: staticcheck // plugins.DataPlugin deprecated
-	return func(*models.DataSource) (plugins.DataPlugin, error) {
-		return &OpenTsdbExecutor{
-			httpClientProvider: httpClientProvider,
-		}, nil
+func NewInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
```

Comment:
This could/should be non-exported since it's only used internally in the package.

```suggestion
func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
```